src/jit/morph.cpp

```diff
@@ -5630,7 +5630,8 @@ GenTreePtr Compiler::fgMorphArrayIndex(GenTreePtr tree)
         // dereference.
         // Also we allocate the temporary when the arrRef is sufficiently complex/expensive.
         //
-        if ((arrRef->gtFlags & (GTF_ASG | GTF_CALL | GTF_GLOB_REF)) || gtComplexityExceeds(&arrRef, MAX_ARR_COMPLEXITY))
+        if ((arrRef->gtFlags & (GTF_ASG | GTF_CALL | GTF_GLOB_REF)) ||
```

Comment:
It's basically due to the complexity of a `GT_FIELD` expansion, and the need for this additional condition was exposed by the change to not set GTF_GLOB_REF.

---
In reply to: [95664107](https://github.com/dotnet/coreclr/pull/8402#discussion_r95664107) [](ancestors = 95664107)